src/_guides/language/numbers.md

```diff
@@ -0,0 +1,547 @@
+  (16-, 32-, or 64-bit).
+  In Java, the `float` and `double` types for fractional values
+  were originally designed to strictly follow IEEE 754 on all platforms,
+  but this constraint was loosened almost immediately for efficiency reasons
+  (`strictfp` is required for exact coherence).
+{{ site.alert.end }}
+
+
+## Dart number representation
+
+In Dart, all numbers are part of the common `Object` type hierarchy,
+and there are two concrete, user-visible numeric types:
+`int`, representing integer values, and `double`, representing fractional values.
+
+```
+  Object
+    |
+   num
+  /    \
+int   double
+```
+***[TODO: Make this a real figure]***
+
+Depending on the platform,
+those numeric types have different, hidden implementations.
+In particular, Dart has two very different types of targets it compiles to:
+
+* **Native:** Most often, a 64-bit mobile or desktop processor.
+* **Web:** JavaScript as the primary execution engine.
+
+The following table shows how Dart numbers are usually implemented:
+
+<table class="table table-striped nowrap">
+  <tr>
+   <th>Representation</th>
+   <th>Native <code>int</code></th>
+   <th>Native <code>double</code></th>
+   <th>Web <code>int</code></th>
+   <th>Web <code>double</code></th>
+  </tr>
+  <tr>
+   <td><a href="https://en.wikipedia.org/wiki/Two%27s_complement">
+     64-bit signed two’s complement</a>
+   </td>
+   <td>✅</td>
+   <td></td>
+   <td></td>
+   <td></td>
+  </tr>
+  <tr>
+   <td>
+     <a href="https://en.wikipedia.org/wiki/Double-precision_floating-point_format">64-bit floating point</a>
+   </td>
+   <td></td>
+   <td>✅</td>
+   <td>✅</td>
+   <td>✅</td>
+  </tr>
+</table>
+
+***[QUESTION: Could we simplify this section? The \* classes are confusing.
+Maybe we could have a single figure for the platform-specific types
+that shows the native & web class hierarchies side by side?
+Then we could just point out, hey look, int & double can be the same type
+in a web program. It might be easier for me to explain if I could see
+examples of the underlying types you're talking about.]***
+
+For native targets, you can assume that
+`int` maps to a signed 64-bit integer representation and
+`double` maps to a 64-bit, IEEE floating-point representation
+that matches the underlying processor:
+
+```
+  Object
+     |
+    num
+   /    \
+ int   double 
+  |       |
+native* native*
+  int   double
+```
+***[TODO: Make this a real figure]***
+
+In general, you can ignore the platform-specific types,
+and think of `int` and `double` as concrete implementation types.
+
+{{ site.alert.info }}
+  **Implementation note:**
+  Dart represents `int` and `double` in
+  a few different ways for efficiency,
+  but these are hidden (marked `*` above) and
+  beyond the scope of this discussion.
+  ***[QUESTION: Can we explain this differently?
+  Drop this note?
+  I don't understand what the \* really means.
+  An example might help.]***
```

Comment:
See above.  Not sure what the best balance to strike here is.  

The primary point I want to make is that an `int` on the web is both an `int` and a `double` - because the concrete type ends up implementing both `int` and `double`.  

In contrast, the concrete type for `int` on native just implements `int`.

Reply:
That helps!